File: sandbox-runner/patterns/mapreduce-tree/go/activities.go
package main

import (
	"context"
	"fmt"
	"time"
)

func ProcessLeaf(_ context.Context, record string) (string, error) {
	// Simulate processing and return a result string.
	time.Sleep(50 * time.Millisecond)
	return fmt.Sprintf("processed(%s)", record), nil
}

File: sandbox-runner/patterns/mapreduce-tree/java/src/main/java/Activities.java
import io.temporal.activity.ActivityInterface;

@ActivityInterface
public interface Activities {
    String processLeaf(String record);

    final class Impl implements Activities {
        @Override
        public String processLeaf(String record) {
            try {
                Thread.sleep(50);
            } catch (InterruptedException e) {
                Thread.currentThread().interrupt();
            }
            return "processed(" + record + ")";
        }
    }
}
